packages/react-components/src/components/gauge/utils/getThresholds.ts

```diff
@@ -1,29 +1,113 @@
-import { getDecimalValueFromMinMax } from './getDecimalValueFromMinMax';
-import { GaugeProps } from '../types';
+import { getThresholdRangeFromMinMax } from './getThresholdRangeFromMinMax';
+import { GaugeProps, GaugeSettings } from '../types';
 import { DEFAULT_GAUGE_PROGRESS_COLOR } from '../constants';
+import { ComparisonOperator } from '@iot-app-kit/core';
 
+type ThresholdRange = {
+  start?: number;
+  end?: number;
+  color: string;
+  comparisonOperator: ComparisonOperator;
+};
+
+// straightforward since they are represented in the same way that echarts represents its colors
+const getColorValuePairsForLTandLTE = (allThresholds: ThresholdRange[]) => {
+  const lineStylesLTandLTE: [number, string][] = [];
+
+  const lteThresholds = allThresholds.filter(
+    (tr) => tr.comparisonOperator === 'LT' || tr.comparisonOperator === 'LTE'
+  );
+
+  lteThresholds.forEach((thresholdRange) => {
+    const { end, color } = thresholdRange;
+    end && lineStylesLTandLTE.push([end, color]);
+  });
+
+  return lineStylesLTandLTE;
+};
+
+// more complicated since GT and GTE are colored in the opposite way that echarts represents its colors
+const getColorValuePairsForGTandGTE = (
+  allThresholds: ThresholdRange[],
+  settings?: GaugeSettings
+) => {
+  const lineStylesGTandGTE: [number, string][] = [];
+  let lastThreshold: ThresholdRange | undefined = undefined;
+
+  // filter thresholds to only be GT or GTE, then sort them from greatest to smallest
+  const gteThresholds = allThresholds
+    .filter(
+      (tr) => tr.comparisonOperator === 'GT' || tr.comparisonOperator === 'GTE'
+    )
+    .sort((a, b) => (b.start ?? 0) - (a.start ?? 0));
+
+  // since the thresholds are sorted in descending order, the first one we see will provide the end range for the second threshold
+  // basically each preceding threshold provides the ceiling for the next one
```

Comment:
Try to keep your code comment lines at 80 characters max to improve readability.

Reply:
Also, prefer /* */ (not /** */ which is a doc comment) instead of // for multi-line comments.